adapters/Unity/UnityAdapter/GADMediationAdapterUnity.m

```diff
@@ -51,9 +49,8 @@ + (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
           gameIDs);
     NSLog(@"Initializing Unity Ads SDK with the game ID %@.", gameID);
   }
-
-  [[GADMAdapterUnitySingleton sharedInstance] initializeWithGameID:gameID];
-  completionHandler(nil);
+  GADMUnityInitializationDelegate* initializationDelegate = [[GADMUnityInitializationDelegate alloc] initWithCompletionHandler:completionHandler];
```

Comment:
Per other comments earlier, it'd be nice if this class can just forward on the block, and expect GADMAdapterUnity to call it at the right time. How GADMAdapterUnity does that is an implementation detail that this class shouldn't need to worry about.